pkgs/by-name/s-/s-search/package.nix

```
@@ -0,0 +1,51 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  testers,
+  s-search,
+  nix-update-script,
+}:
+
+buildGoModule (finalAttrs: {
+  pname = "s-search";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "zquestz";
+    repo = "s";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-cu2XGA5rRO+cHZ9QkJvMBjbxNZigxsx12ziCJG1LCQs=";
+  };
+
+  vendorHash = "sha256-0E/9fONanSxb2Tv5wKIpf1J/A6Hdge23xy3r6pFyV9E=";
+
+  ldflags = [
+    "-s"
+    "-w"
+  ];
+
+  passthru = {
+    # `versionCheckHook` fails due to the program requires `sh` to be available in `PATH`
+    tests.version = testers.testVersion {
+      package = s-search;
```

It's bad practice because the test will not respect any overrides. Instead, use the package after overrides are applied:

```suggestion
      package = finalAttrs.finalPackage;
```